File: vigia_ui/lib/domain/enums/device_rooms.dart
enum DeviceRooms {
  bedroom,
  livingRoom,
  kitchen,
  bathroom,
  office,
  garage,
  backyard,
  frontyard,
}

extension DeviceRoomsExtension on DeviceRooms {
  String toApiString() {
    return switch (this) {
      DeviceRooms.bedroom => 'Bedroom',
      DeviceRooms.livingRoom => 'LivingRoom',
      DeviceRooms.kitchen => 'Kitchen',
      DeviceRooms.bathroom => 'Bathroom',
      DeviceRooms.office => 'Office',
      DeviceRooms.garage => 'Garage',
      DeviceRooms.backyard => 'Backyard',
      DeviceRooms.frontyard => 'Frontyard',
    };
  }

  static DeviceRooms? fromString(String? value) {
    if (value == null) return null;

    switch (value) {
      case 'Bedroom':
        return DeviceRooms.bedroom;
      case 'LivingRoom':
        return DeviceRooms.livingRoom;
      case 'Kitchen':
        return DeviceRooms.kitchen;
      case 'Bathroom':
        return DeviceRooms.bathroom;
      case 'Office':
        return DeviceRooms.office;
      case 'Garage':
        return DeviceRooms.garage;
      case 'Backyard':
        return DeviceRooms.backyard;
      case 'Frontyard':
        return DeviceRooms.frontyard;
      default:
        return null;
    }
  }
}

File: vigia-web/src/app/core/enums/device-rooms.ts
export enum DeviceRooms {
  Bedroom = 'Bedroom',
  LivingRoom = 'LivingRoom',
  Kitchen = 'Kitchen',
  Bathroom = 'Bathroom',
  Office = 'Office',
  Garage = 'Garage',
  Backyard = 'Backyard',
  Frontyard = 'Frontyard',
}

const DEVICE_ROOM_VALUES = new Set<string>(Object.values(DeviceRooms));

export function parseDeviceRoom(value: string | null | undefined): DeviceRooms | null {
  if (!value || !DEVICE_ROOM_VALUES.has(value)) {
    return null;
  }
  return value as DeviceRooms;
}

export function deviceRoomI18nKey(room: DeviceRooms | null | undefined): string {
  if (!room) {
    return 'DEVICES.ROOMS.NotDefined';
  }
  return `DEVICES.ROOMS.${room}`;
}
